apps/api/src/services/competition-manager.service.ts

```diff
@@ -486,14 +508,15 @@ export class CompetitionManager {
 
       // Fallback to calculating current values
       const agents = await getCompetitionAgents(competitionId);
-      const leaderboard: { agentId: string; value: number }[] = [];
+      const leaderboard: { agentId: string; value: number; pnl: number }[] = [];
 
       for (const agentId of agents) {
         const portfolioValue =
           await this.tradeSimulator.calculatePortfolioValue(agentId);
         leaderboard.push({
           agentId,
           value: portfolioValue,
+          pnl: 0, // TODO: if there's no competitions_leaderboard row we don't have a pnl
```

Comment:
I didn't want to add any more to this PR, but we might want to handle this differently in the future.